File: Java/Guias/guia4/ej9_BankAccount/CheckingAccount.java
package Guias.guia4.ej9_BankAccount;

public class CheckingAccount extends BankAccount {
    private final double overdraft;

    public CheckingAccount(int id, double over) {
        super(id);
        this.overdraft = over;
    }

    @Override
    protected boolean canExtract(double amount) {
        return amount - balance <= overdraft;
    }

    public double getOverdraft() {
        return this.overdraft;
    }
}

File: RUBY/Guias/Guia8/Ej5_CheckingAccount/checking_account.rb
# frozen_string_literal: true

require_relative 'bank_account'
class CheckingAccount < BankAccount
  def initialize(id, overdraft)
    create(id)
    @overdraft = overdraft
  end

  private def canExtract?(amount)
    amount >= @saldo + @overdraft
  end

end
